plugins/detabler/plugin.js

```diff
@@ -1,9 +1,69 @@
 'use strict';
 
-(function() {
-    CKEDITOR.plugins.add( 'detabler', {    
-    init: function( editor ) {
-        // Plugin logic goes here...
-    }
-});
-})();
\ No newline at end of file
+(function () {
+    CKEDITOR.plugins.add('detabler', {
+		requires: 'table,contextmenu',
+		init: function (editor) {
+			editor.addCommand('detable', {
+				exec: function (editor) {
+					var path,
+						table,
+						cells,
+						currentCell,
+						currentItem;
+
+					path = editor.elementPath();
+					table = path.contains('table');
+
+					if (table) {
+						table.$.normalize(); // we don't want to deal with un-merged text nodes here.
+						cells = table.find('td, th');
+						// loop through all cell-like items
+						for (var i = 0; i < cells.count(); i++) {
```

Comment:
Ha, that's funny.  Believe it or not this habit goes back to the C++ textbook we used for programming in high school....